web/src/components/storage/ProposalResultSection.jsx

```diff
@@ -0,0 +1,286 @@
+import DevicesManager from "~/components/storage/DevicesManager";
+import { If, Section, Reminder, Tag, TreeTable } from "~/components/core";
+import { ProposalActionsDialog } from "~/components/storage";
+
+/**
+ * @typedef {import ("~/client/storage").Action} Action
+ * @typedef {import ("~/client/storage").StorageDevice} StorageDevice
+ * @typedef {import("~/client/mixins").ValidationError} ValidationError
+ */
+
+/**
+ * Renders information about planned actions, allowing to check all of them and warning with a
+ * summary about the deletion ones, if any.
+ * @component
+ *
+ * @param {object} props
+ * @param {Action[]} props.actions
+ * @param {string[]} props.systems
+ */
+const DeletionsInfo = ({ actions, systems }) => {
+  const total = actions.length;
+
+  if (total === 0) return;
+
+  // TRANSLATORS: %d will be replaced by the amount of destructive actions
+  const warningTitle = sprintf(n_(
+    "There is %d destructive action planned",
+    "There are %d destructive actions planned",
+    total
+  ), total);
+
+  // FIXME: Use the Intl.ListFormat instead of the `join(", ")` used below.
+  // Most probably, a `listFormat` or similar wrapper should live in src/i18n.js or so.
+  // Read https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat
+  return (
+    <Reminder title={warningTitle} variant="subtle">
+      <If
+        condition={systems.length > 0}
+        then={
+          <p>
+            {_("Including the deletion of")} <strong>{systems.join(", ")}</strong>
+          </p>
+        }
+      />
+    </Reminder>
+  );
+};
+
+/**
+ * Renders needed UI elements to allow user check the proposal planned actions
+ * @component
+ *
+ * @param {object} props
+ * @param {Action[]} props.actions
+ */
+const ActionsInfo = ({ actions }) => {
+  const [showActions, setShowActions] = useState(false);
+  const onOpen = () => setShowActions(true);
+  const onClose = () => setShowActions(false);
+
+  return (
+    <>
+      <Button onClick={onOpen} variant="link" isInline>{_("Check all planned actions")}</Button>
+      <ProposalActionsDialog actions={actions} isOpen={showActions} onClose={onClose} />
+    </>
+  );
+};
+
+/**
+ * Renders a TreeTable rendering the devices proposal result.
+ * @component
+ *
+ * @param {object} props
+ * @param {DevicesManager} props.devicesManager
+ */
+const DevicesTreeTable = ({ devicesManager }) => {
+  const renderDeviceName = (item) => {
+    let name = item.sid && item.name;
+    // NOTE: returning a fragment here to avoid a weird React complaint when using a PF/Table +
+    // treeRow props.
+    if (!name) return <></>;
+
+    if (["partition", "lvmLv"].includes(item.type))
+      name = name.split("/").pop();
+
+    return (
+      <div className="split">
+        <span>{name}</span>
+      </div>
+    );
+  };
+
+  const renderNewLabel = (item) => {
+    if (!item.sid) return;
+
+    // FIXME New PVs over a disk is not detected as new.
```

Comment:
We would need a Trello card collecting all the shortcuts taken and the corner-cases found on this pull request.

Reply:
Thinking it twice, I don't think that can ever happen. The proposal only create PVs on partitions. The `GuidedProposal` is partition-driven. It can reuse a given VG, LV or partition, but it will never turn a block device into a PV without partitioning it.